### PR TITLE
Fix recommended Kubernetes version in README

### PR DESCRIPTION
**Pull request by chrishenzie**
**What type of PR is this?**
/kind cleanup
/kind documentation

**What this PR does / why we need it**:
Fixes an issue in the README where the recommended Kubernetes version is set to one higher than it should be.

See for more details:
https://kubernetes-csi.github.io/docs/project-policies.html#recommended-version

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @xing-yang

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,9 +22,9 @@ This information reflects the head of this branch.
...
-| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | [CSI Spec v1.5.0](https://github.com/container-storage-interface/spec/releases/tag/v1.5.0) | k8s.gcr.io/sig-storage/snapshot-validation-webhook  | 1.20     | 1.22         |
+| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | [CSI Spec v1.5.0](https://github.com/container-storage-interface/spec/releases/tag/v1.5.0) | k8s.gcr.io/sig-storage/csi-snapshotter | 1.20         | 1.20         |
+| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | [CSI Spec v1.5.0](https://github.com/container-storage-interface/spec/releases/tag/v1.5.0) | k8s.gcr.io/sig-storage/snapshot-controller  | 1.20     | 1.20         |
+| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | [CSI Spec v1.5.0](https://github.com/container-storage-interface/spec/releases/tag/v1.5.0) | k8s.gcr.io/sig-storage/snapshot-validation-webhook  | 1.20     | 1.20         |
```
Comment (chrishenzie) [inline README.md:27]: @xing-yang By this same logic should I reduce the recommended CSI spec version?

Comment (xing-yang) [reply README.md:27]: I think CSI is okay.  CSI spec 1.0 is too old.  @msau42 What do you think?  It's confusing.

Comment (msau42) [reply README.md:27]: updating csi spec to 1.5 seems fine.

Comment (xing-yang) [reply README.md:27]: Ok

**Comment by chrishenzie** [comment]
Holding so we can clarify https://github.com/kubernetes-csi/external-provisioner/pull/661#discussion_r683066993 before proceeding.

**Comment by msau42** [comment]
For snapshotter, it's a little subtle, but since access modes is not used in the snapshotter I think keeping it at 1.20 is fine?x

**Comment by chrishenzie** [comment]
Yeah they aren't used here, thanks for catching

/hold cancel

**Pull request merged**